### PR TITLE
Add mutation from `foo[bar] || baz` to `foo.fetch(bar) { baz }`

### PR DESCRIPTION
**Pull request by backus**

### CODE REVIEW

**Comment by backus** [comment]
@mbj this is ready for review

File: lib/mutant/mutator/node/binary.rb [hunk 1]
```diff
@@ -60,6 +61,21 @@ def emit_left_negation
           emit(s(node.type, n_not(left), right))
         end
 
+        # Emit mutation from `a[b] || c` to `a.fetch(b) { c }`
+        #
+        # @return [undefined]
+        def emit_fetch_mutation
+          send_node = AST::Meta::Send.new(left)
```
Comment (mbj) [inline lib/mutant/mutator/node/binary.rb]: Instantiating `AST::Meta::Send` is guarded to only happen for `left` nodes of type `:send`? It should.

Comment (backus) [inline lib/mutant/mutator/node/binary.rb]: Mutant kept removing it and I'm not sure I can think of another node that might have `:[]` as the second positional argument in the node (making it the `#selector` attribute under the meta)

Comment (mbj) [inline lib/mutant/mutator/node/binary.rb]: This is a "type check by AST structure", I've to think about counter examples, but from a first glance I think you are right.

Comment (backus) [inline lib/mutant/mutator/node/binary.rb]: @mbj I ended up adding a test case that used `csend` which justifies including `n_send?`

Comment (mbj) [inline lib/mutant/mutator/node/binary.rb]: We should not create `send_node` that assumes `left` is `:send` before the node type check.

Comment (backus) [inline lib/mutant/mutator/node/binary.rb]: k

Comment (backus) [inline lib/mutant/mutator/node/binary.rb]: @mbj fixed

**Comment by backus** [comment]
@mbj ready for another pass

**Comment by mbj** [comment]
Closing this PR as all contributions will require a CLA from now on.

**Pull request closed without merging**